src/panels/lovelace/editor/card-editor/hui-edit-card.ts

```diff
@@ -86,6 +87,22 @@ export class HuiEditCard extends LitElement {
     return this.shadowRoot!.querySelector("hui-card-preview")!;
   }
 
+  private get _cardJSON(): LovelaceCardConfig | undefined {
+    return this._cardConfig;
+  }
+
+  private set _cardJSON(config: LovelaceCardConfig | undefined) {
+    this._cardConfig = config;
+  }
+
+  private get _cardYAML(): string | undefined {
+    return this.cardConfig ? yaml.safeDump(this._cardJSON) : undefined;
```

Comment:
So the reason we didn't do this was to save the YAML formatting, especially keys can get re-sorted if you do YAML -> JSON -> YAML.

Reply:
You mean we don't want to do it right away? Because we are doing that in the end when saving

Reply:
Correct

Reply:
YAML has different formats that can represent the same thing. We should avoid "patching" how it looks to not confuse users. We do this for JSON in the automation editor, the moment you make it valid JSON, the text all jumps around.

Reply:
Yes. This caused some problems in the yaml editor, e.g. adding a new key would automatically give it the value `null` and move the insertion point to the first line...

A fix will be pushed soon